src/runtime/pybuffer.cs

```diff
@@ -0,0 +1,228 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Python.Runtime
+{
+    public class PyBuffer : IDisposable
+    {
+        private PyObject _exporter;
+        private IntPtr _handle;
+        private Runtime.Py_buffer _view;
+        private GCHandle _gchandle;
+
+        unsafe internal PyBuffer(PyObject exporter, int flags)
+        {
+            int size = Marshal.SizeOf(typeof(Runtime.Py_buffer));
+            byte[] rawData = new byte[size];
+            _gchandle = GCHandle.Alloc(rawData, GCHandleType.Pinned);
+            _handle = _gchandle.AddrOfPinnedObject();
+
+            if (Runtime.PyObject_GetBuffer(exporter.Handle, _handle, flags) < 0)
```

Comment:
Why do yo need to keep `_handle` and `_gchandle`? Can't you simply pass address of `_view` where a pointer to `Py_buffer` is expected?